### PR TITLE
Add `npcbossbar` command

### PR DESCRIPTION
**Pull request by tal5**
## Additions

- `npcbossbar` - a command to configure an NPC's bossbar trait.

## Notes

- Not sure if `progress` is the best name there? as `health`/placeholders are a valid input as well, and `progress:health` is a little weird.

### CODE REVIEW

File: plugin/src/main/java/com/denizenscript/denizen/scripts/commands/npc/NPCBossBarCommand.java [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+    // @Required 1
+    // @Maximum 8
+    // @Short Controls or removes the linked NPC's bossbar.
+    // @Group npc
+    //
+    // @Description
+    // Controls or removes the linked NPC's bossbar.
+    //
+    // Progress can be a number between 1 and 100 or 'health' to make it track the NPC's health.
+    // Placeholder API/Citizens placeholders are supported
```
Comment (mcmonkey4eva) [inline plugin/src/main/java/com/denizenscript/denizen/scripts/commands/npc/NPCBossBarCommand.java]: period at end of sentence

File: plugin/src/main/java/com/denizenscript/denizen/scripts/commands/npc/NPCBossBarCommand.java [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+    // # Removes a specific NPC's bossbar.
+    // - npcbossbar remove npc:<[theNPC]>
+    // -->
+
+    @Override
+    public void addCustomTabCompletions(TabCompletionsBuilder tab) {
+        tab.addWithPrefix("color:", BarColor.values());
+        tab.addWithPrefix("options:", BarFlag.values());
+        tab.addWithPrefix("style:", BarStyle.values());
+    }
```
Comment (mcmonkey4eva) [inline plugin/src/main/java/com/denizenscript/denizen/scripts/commands/npc/NPCBossBarCommand.java:69]: should have `progress:health` tab completed

**Comment by mcmonkey4eva** [comment]
`Not sure if progress is the best name there? as health/placeholders are a valid input as well, and progress:health is a little weird.`
I believe in Citizens it's called `track`
I don't think there's a great naming either way, as it's multiple different types of input that are all valid to the one spot - you make one sound right, the other will sound wrong

File: plugin/src/main/java/com/denizenscript/denizen/scripts/commands/npc/NPCBossBarCommand.java (hunk 1 above)
Comment (tal5) [reply plugin/src/main/java/com/denizenscript/denizen/scripts/commands/npc/NPCBossBarCommand.java]: Fixed

File: plugin/src/main/java/com/denizenscript/denizen/scripts/commands/npc/NPCBossBarCommand.java (hunk 2 above)
Comment (tal5) [reply plugin/src/main/java/com/denizenscript/denizen/scripts/commands/npc/NPCBossBarCommand.java:69]: Added

**Comment by tal5** [comment]
> `Not sure if progress is the best name there? as health/placeholders are a valid input as well, and progress:health is a little weird.` I believe in Citizens it's called `track` I don't think there's a great naming either way, as it's multiple different types of input that are all valid to the one spot - you make one sound right, the other will sound wrong

Yeah I guess you're right, will keep it as `progress` for now to match the `bossbar` command and as it'd probably mostly be used for that (and not for PAPI placeholders or whatnot)

**Pull request merged**